### ISSUE TITLE
Run specific test by fully qualified name

### ISSUE THREAD

**Issue by mchernyavsky** [rust-lang/rust#55623]
Suppose I have two test targets:

1. tests/tests1.rs
```rust
#[test]
fn test1() {
    panic!()
}

#[test]
fn test2() {}
```

2. tests/tests2.rs
```rust
#[test]
fn test1() {}

#[test]
fn test2() {
    panic!()
}
```

I want to run only test1 from tests1 and test2 from tests2. The `--exact` option doesn't help me in this case.

I want to have some option to run a specific test, like this:
```
cargo test --no-fail-fast --test tests1 --test tests2 -- --fqn tests1::test1 --fqn tests2::test2
```

**Comment by Enselic** [comment]
Triage: Can someone with sufficient privileges move this to the cargo project please? Thanks!

**Comment by ehuss** [comment]
I don't think this is something cargo can resolve on its own, since the test filtering is performed in libtest (which lives in this repo), and cargo doesn't know which tests live in which crate. 

I'll cc @epage in case he knows if this is already tracked somewhere, or if it would be on the roadmap for testing-devex.

**Comment by epage** [comment]
Everything tagged `A-libtest` automatically gets added to testing-devex backlog, so this will be on our radar.

For use cases, I'm interested in know why to hand pick individual tests but not hand pick the test targets at the time.  I'm assuming it was these two tests that failed.  

Some options
- Going in a different direction than this solution, a rerun last failed (which doesn't seem to be in the backlog)
- In a custom test harness, we add the target name to the test name (one route we'll be exploring is shifting focus away from libtest to custom test harnesses with a premier harness we'll be developing)
- Add a new flag to the harness as proposed here, whether in libtest (unlikely) or in the standard extensions for custom test harnesses
